monkey/infection_monkey/exploit/sshexec.py

```diff
@@ -110,47 +120,72 @@ def _exploit_host(self):
 
         is_open, _ = check_tcp_port(self.host.ip_addr, port)
         if not is_open:
-            logger.info("SSH port is closed on %r, skipping", self.host)
-            return False
+            exploit_result["error_message"] = f"SSH port is closed on {self.host}, skipping"
+
+            logger.info(exploit_result["error_message"])
+            return self._get_exploit_result_data(exploit_result)
 
         try:
             ssh = self.exploit_with_ssh_keys(port)
+            exploit_result["exploitation_success"] = True
         except FailedExploitationError:
             try:
                 ssh = self.exploit_with_login_creds(port)
+                exploit_result["exploitation_success"] = True
             except FailedExploitationError:
-                logger.debug("Exploiter SSHExploiter is giving up...")
-                return False
+                exploit_result["error_message"] = "Exploiter SSHExploiter is giving up..."
+                exploit_result["exploitation_success"] = False
+                exploit_result["propagation_success"] = False
+
+                logger.debug(exploit_result["error_message"])
+                return exploit_result
 
         if not self.host.os.get("type"):
             try:
-                _, stdout, _ = ssh.exec_command("uname -o")
+                _, stdout, _ = ssh.exec_command("uname -o")  # propagation is True
                 uname_os = stdout.read().lower().strip().decode()
+                exploit_result["propagation_success"] = True
```

Comment:
I don't think we can say propagation is true. We're just executing `uname -o`. We haven't successfully launched a monkey yet.